testutil/base.go

```diff
@@ -60,3 +62,31 @@ func (s *BaseTest) TearDownTest(c *check.C) {
 func (s *BaseTest) AddCleanup(f func()) {
 	s.cleanupHandlers = append(s.cleanupHandlers, f)
 }
+
+// Mocks any mutable object with a compliant substitute, and insures it will be restored after the test
+func (s *BaseTest) MockAny(mockedByPtr interface{}, mock interface{}) {
+	restore := MockAny(mockedByPtr, mock)
+	s.AddCleanup(restore)
+}
+
+// Mocks any mutable object with a compliant substitute, and returns a restore function
+func MockAny(mockedByPtr interface{}, mock interface{}) (restore func()) {
+	mockedPtr, mockValue := reflect.ValueOf(mockedByPtr), reflect.ValueOf(mock)
+
+	if mockedPtr.Type().Kind() != reflect.Ptr {
+		panic("Mock: first argument shall be a pointer to the mocked element!")
```

Comment:
How about `MockAny: ...` ?

Reply:
Do you mean to start the comment with the name of the API?
You are totally right. I'm just still learning to get used to it. It has totally slipped my mind.

Thank you. :pray:

Reply:
Yes, I mean to start the panic message with "MockAny: <error message>".